pydantic/json_schema.py

```diff
@@ -1112,31 +1112,41 @@ def default_schema(self, schema: core_schema.WithDefaultSchema) -> JsonSchemaVal
         if default is NoDefault:
             return json_schema
 
+        def _get_ser_schema(schema: CoreSchema) -> core_schema.PlainSerializerFunctionSerSchema | None:
+            if (
+                (ser_schema := schema.get('serialization'))
+                and ser_schema['type'] == 'function-plain'
+                and not ser_schema.get('info_arg')
+            ):
+                return ser_schema
+            if _core_utils.is_function_with_inner_schema(schema):
+                return _get_ser_schema(schema['schema'])
```

Comment:
Looks like this function doesn't capture any state from the surrounding body, maybe move it to the top level to avoid the perf hit of creating a new function object each time?